### ISSUE TITLE
Confusing predictions when implementing GoogleNet

### ISSUE THREAD

**Issue by aaronpolhamus** [BVLC/caffe#4133]
I'm user the Caffe implementation of GoogleNet available from https://github.com/BVLC/caffe/tree/master/models/bvlc_googlenet in a detection and localization process. I've had great success using this network for transfer learning on my own fine-grained classification task, but I've had a much harder time using the out-of-the-box implementation. Using a slightly tweaked version of [deploy.prototxt](https://gist.github.com/aaronpolhamus/5e8abf4cf5cb1ff11e5cf4ecba4afd16) with batch sized reduced from 10 to 1 I'm predicting the classes for the images [contained in this directory](https://www.dropbox.com/sh/s9o83sngobrqdpy/AAAmzBcIZ_VEtTXOhn_C9oKRa?dl=0). 

This results in the following top-5 probabilities: 

    {
    'bikini1': {
        '445': 0.44777653,
        '459': 0.42523962,
        '543': 0.0014282113,
        '638': 0.083084606,
        '639': 0.03712019},
    'bikini2': {
        '445': 0.34988937,
        '459': 0.56838924,
        '638': 0.053508826,
        '639': 0.0262988,
        '775': 0.00064279145},
    'cheeseburger1': {
        '923': 0.00077056035,
        '924': 0.00074912253,
        '931': 0.0040806201,
        '933': 0.98862922,
        '964': 0.0012325585},
    'cheeseburger2': {
        '923': 5.947132e-05,
        '928': 0.00013152717,
        '931': 0.00084521802,
        '933': 0.99827468,
        '934': 0.00013743658},
    'ipod1': {
        '487': 0.0010906343,
        '605': 0.99825007,
        '632': 9.7395357e-05,
        '662': 0.00010778452,
        '754': 0.00011993563},
    'ipod2': {
        '487': 1.3061179e-05,
        '592': 2.9665862e-05,
        '605': 0.99993002,
        '681': 9.4235074e-06,
        '761': 3.5998178e-06}
    }

These images are taken directly from the ImageNet training set. What's interesting is that the network assigns the same label to each pair with a high degree of confidence, but these labels seem totally at odds with the [labels provided by ImageNet](https://gist.github.com/aaronpolhamus/964a4411c0906315deb9f4a3723aac57). The network seems to be working, but maybe I'm referencing the wrong class-label mapping? Any help from all you Caffe/ImageNet specialists would be greatly appreciated!

**Comment by aaronpolhamus** [comment]
knew it had to be out there somewhere. thanks @shaibagon. http://stackoverflow.com/questions/27986339/where-can-i-find-the-label-map-between-trained-model-like-googlenets-output-to/28009396#28009396
